packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -52,7 +56,7 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
           if (!s) s = new MagicString(code)
 
           // potential dynamic template string
-          if (rawUrl[0] === '`' && /\$\{/.test(rawUrl)) {
+          if (rawUrl[0] === '`' && placeholderRE.test(rawUrl)) {
```

Comment:
Maybe we can use `rawUrl.includes('${')` instead?

[simple benchmark](https://perf.link/#eyJpZCI6IjR4cmpubmgzcjRoIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGZvbyA9IFwiYVwiLnJlcGVhdCgxMCkgKyBcIiR7Zm9vYmFyfVwiICsgXCJhXCIucmVwZWF0KDEwKSIsInRlc3RzIjpbeyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImZvby5pbmNsdWRlcygnJHsnKSIsInJ1bnMiOlsyNjMzMDAwLDIyODIwMDAsMTM0MzAwMCwyMjE0MDAwLDQyMjYwMDAsMjQwNTAwMCwyNjkzMDAwLDExMjEwMDAsMzU3MDAwMCwyMzMyMDAwLDMxMDAwLDcxOTAwMCw1NzAwMDAsMjU2NDAwMCw5NzIwMDAsMjkzNDAwMCwxNjcwMDAwLDI1NDMwMDAsODA2MDAwLDEwMDAsMjc1NzAwMCwyODE3MDAwLDQzODkwMDAsMTYwNzAwMCwyNzEzMDAwLDI5MzUwMDAsMjg1MTAwMCwxMTcwMDAwLDUwMDAsMjU5ODAwMCw1NjgwMDAsMjY0NDAwMCw4ODAwMDAsMTA4NjAwMCwxNjM5MDAwLDI5MDgwMDAsNDY0MDAwLDIxMDEwMDAsMjYxNTAwMCwxMTA5MDAwLDI2MzQwMDAsMTk0NjAwMCw2NjMwMDAsMjAxNDAwMCwyNTYwMDAwLDE3NDEwMDAsMjIxNTAwMCwxMzE4MDAwLDE3MTQwMDAsMzk5NjAwMCwyNzg5MDAwLDI2MTkwMDAsMzI1MzAwMCwzMzU3MDAwLDE5OTcwMDAsMjMyNjAwMCwzMTcwMDAsMjMyMzAwMCwzMTkxMDAwLDQ0NzAwMCwxMzY4MDAwLDQ2MDAwLDIxNDcwMDAsMzA2NjAwMCwxNTQ4MDAwLDIwMDEwMDAsMTEwMDAwLDEyNDkwMDAsMjA0MDAwLDQzMjgwMDAsOTA3MDAwLDIxODAwMCwzMTA5MDAwLDExMzcwMDAsMjYzODAwMCwxNzc2MDAwLDE2NzAwMDAsODI1MDAwLDkyNzAwMCwxNjgzMDAwLDQzODQwMDAsMTM3NDAwMCwzMzIwMDAsMjc1OTAwMCwyMTgzMDAwLDgwMDAwLDE2MTAwMCwxMjM4MDAwLDEzMTgwMDAsNjM0MDAwLDcwMDAsNjUzMDAwLDE3NTUwMDAsMTExNjAwMCwxNjk0MDAwLDIyMDAwMDAsMjA2OTAwMCwxNTEwMDAwLDEwNjgwMDAsMTcwMDAwXSwib3BzIjoxNzg1NjcwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiL1xcJFxcey8udGVzdChmb28pIiwicnVucyI6WzU1MDAwLDM0OTEwMDAsMTA2OTAwMCwxMDYwMDAsMzU0MzAwMCw5ODUwMDAsMTQwNDAwMCwxMzgyMDAwLDEwNDYwMDAsMzcyNDAwMCw3MjYwMDAsMjA1NDAwMCw0NjMwMDAsMTk3OTAwMCwzMDIwMDAsNzMxMDAwLDE2NDgwMDAsMjc0MDAwMCwzOTAyMDAwLDYxNDAwMCwxNzAwMCw1ODUwMDAsNTI4MDAwLDE3MTQwMDAsMTU0ODAwMCwzMTcwMDAwLDExNDAwMCwyMTUwMDAsMTY3MDAwMCwyMzg2MDAwLDI3MDAwMCwxNzA1MDAwLDI0NjcwMDAsMTM0NTAwMCwyNTU4MDAwLDM2MTMwMDAsNTYwMDAsNjI0MDAwLDE3MDAwLDUyNTAwMCwyMzE2MDAwLDk0ODAwMCwxMTM2MDAwLDE4NDkwMDAsMjEwMTAwMCwxNzE0MDAwLDMyMDAwLDI2MDYwMDAsMjExMjAwMCwxNjc5MDAwLDIxMDYwMDAsMzkwMDAsNDgzMDAwLDE5MTIwMDAsMjIwOTAwMCwxOTEwMDAwLDg1OTAwMCwxMDAwLDE2MTYwMDAsMTY0ODAwMCwyMjc5MDAwLDExMDAwLDE5MTgwMDAsMjU0MDAwMCwyNDAyMDAwLDk4MjAwMCw4NDUwMDAsMjUwNTAwMCw1MTAwMDAsMTkwMDAsMzc0MDAwLDk0MDAwLDI2NzIwMDAsMTQ4OTAwMCw4ODAwMDAsMTQzMTAwMCwyMDcxMDAwLDY1MjAwMCwxODc4MDAwLDEyODkwMDAsMjE4MzAwMCwyNDY3MDAwLDMzNTgwMDAsMTcyMDAwMCwxNTQ2MDAwLDE2NTUwMDAsMTk1MjAwMCwxNjQ3MDAwLDE3NzkwMDAsMTEzNzAwMCwxNDE5MDAwLDIwODMwMDAsMTAzMjAwMCwxNDMyMDAwLDIyNDEwMDAsMTM1MTAwMCwxODgyMDAwLDExNTYwMDAsMTQ5NTAwMCwyNDI3MDAwXSwib3BzIjoxNDkxNzAwfV0sInVwZGF0ZWQiOiIyMDIzLTAzLTIxVDAxOjMzOjQyLjY1MloifQ%3D%3D)

Reply:
I would also expect this to be faster, but if the regex is extracted it isn't clear: [benchmark](https://perf.link/#eyJpZCI6IjR4cmpubmgzcjRoIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGZvbyA9IFwiYVwiLnJlcGVhdCgxMCkgKyBcIiR7Zm9vYmFyfVwiICsgXCJhXCIucmVwZWF0KDEwKVxuY29uc3QgcGxhY2Vob2xkZXJSRSA9IC9cXCRcXHsvIiwidGVzdHMiOlt7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiZm9vLmluY2x1ZGVzKCckeycpIiwicnVucyI6WzIwMDAwLDkwMTAwMCwzNTU5MDAwLDE1MzIwMDAsNTA3NTAwMCwxNjE1MDAwLDU2OTUwMDAsODIzOTAwMCw3NzQzMDAwLDczOTkwMDAsNTEzNzAwMCw0NDQwMDAsODYxNTAwMCw1Nzg0MDAwLDQ2NjUwMDAsNzYzNTAwMCw0MDIwMDAsMTM1NTAwMCw2Nzc5MDAwLDU4NzcwMDAsNzQ0MjAwMCw2NTgwMDAwLDIyODUwMDAsNTM2MDAwLDg0MzAwMCwyNTQ5MDAwLDU3MDAwMCwxNjgwMDAsNjc3OTAwMCw4MTAwMCwyOTMwMDAwLDIwNTUwMDAsMjA4MDAwMCwzMTYxMDAwLDUzNzAwMCwyMjA4MDAwLDMyMTAwMCw3OTU4MDAwLDQyMTUwMDAsNjgwMDAwLDE4ODkwMDAsODYxMDAwLDM1ODEwMDAsMzI1MjAwMCw4ODgzMDAwLDU0MjgwMDAsNTQyODAwMCw3MzcxMDAwLDczNDIwMDAsNzI1OTAwMCwxMTY1MDAwLDIyOTMwMDAsODMxMzAwMCw2MTI4MDAwLDE0ODYwMDAsNjg0MDAwLDE0NTMwMDAsNTI3MDAwMCw4ODU4MDAwLDE5MDAwLDE4NjYwMDAsNzUwMDAsMjA1ODAwMCwxMDEwMDAwLDE4OTcwMDAsMTM2MDAwMCw4MTAzMDAwLDg1MjUwMDAsNzQ1OTAwMCw1NDI4MDAwLDY3NDAwMCw4MDQ2MDAwLDcxNzAwMDAsNzU2NDAwMCwzNTMxMDAwLDIzMDkwMDAsODI5MzAwMCwyNjUzMDAwLDI0MTYwMDAsOTE5MDAwLDE3OTMwMDAsNjE0NTAwMCwzNTQ2MDAwLDIzNDAwMCw5MjYwMDAsNjc3OTAwMCwyMTc0MDAwLDg2MTAwMCw3MDMwMDAsNzcxOTAwMCw0MTAzMDAwLDEzOTMwMDAsODA5ODAwMCwyNTgyMDAwLDE4NTYwMDAsMjY3OTAwMCw4MDI1MDAwLDIwMDQwMDAsMTQxOTAwMCwyMzUwMDBdLCJvcHMiOjM3NDA0MjB9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJwbGFjZWhvbGRlclJFLnRlc3QoZm9vKSIsInJ1bnMiOlszMTIxMDAwLDM0MDMwMDAsMzgyMjAwMCw1MDI3MDAwLDE2MjIwMDAsNjAyOTAwMCwxMDQ4MDAwLDU3MTgwMDAsNDEwOTAwMCwzODk3MDAwLDIyMTYwMDAsNjAzNzAwMCw2MjI5MDAwLDM1OTcwMDAsNDI5NDAwMCw2MDAzMDAwLDUyOTkwMDAsNzE4MDAwMCw2NDUwMDAwLDM4NzgwMDAsMzE1MzAwMCw0NzkwMDAwLDUwMjcwMDAsNjQxMTAwMCw1NzA5MDAwLDc3NzUwMDAsNjE4MTAwMCw3NTAxMDAwLDU0MTUwMDAsODE3ODAwMCw3NDMwMDAsMjM2MDAwLDcyMzAwMCw4NzAwMDAsNjE2MTAwMCw3OTk1MDAwLDM3NTYwMDAsNTY4ODAwMCwyMzk1MDAwLDcxNzAwMDAsNzAwMzAwMCw2MTc4MDAwLDg4NjAwMCw3MDAzMDAwLDY4ODUwMDAsMzUwMjAwMCwyMzQ5MDAwLDUzNTIwMDAsNTcxMTAwMCw0NzE5MDAwLDUwMjYwMDAsNTgwODAwMCw2MDkyMDAwLDIxODcwMDAsNzgxNzAwMCw1MDYwMDAwLDY5NzUwMDAsMjcwNjAwMCw1MDczMDAwLDQzMTQwMDAsNjA2NzAwMCw2MzE3MDAwLDUzMDMwMDAsNjExNjAwMCw3NDc4MDAwLDU0NzIwMDAsNTM0MDAwMCw1MTA5MDAwLDQwODMwMDAsMzcxMjAwMCw2NzAyMDAwLDU0NjAwMDAsMzkxMjAwMCw1MjU4MDAwLDIwNzkwMDAsMjU3MDAwLDYyNzkwMDAsNTAyNzAwMCw4MTgyMDAwLDYyMTEwMDAsNjY5NjAwMCwyNzkwMDAsNzczMDAwMCw2MDg4MDAwLDY4ODcwMDAsMTIzMDAwLDIwMDAsNzE4NjAwMCw1MDI3MDAwLDUyNzAwMDAsMzA1OTAwMCw1MjkzMDAwLDU1NjMwMDAsNTIyNzAwMCw1MTkxMDAwLDc5MDAwMCw1NDY5MDAwLDE3MzcwMDAsNzIxODAwMCw2Mzk5MDAwXSwib3BzIjo0ODExMDAwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiL1xcJFxcey8udGVzdChmb28pIiwicnVucyI6WzEwNTkwMDAsODE2MDAwLDE4NzAwMCwzNDg1MDAwLDc3ODkwMDAsMzQzMzAwMCwyMzcwMDAsMjgyNzAwMCwxNjI0MDAwLDM0MjAwMDAsNzAyNTAwMCwzNjg3MDAwLDQ0ODgwMDAsMTAyMjAwMCwyNjE3MDAwLDExNTAwMDAsMTc0NTAwMCwzMTAxMDAwLDU3MzgwMDAsMjUwNzAwMCwyMTE3MDAwLDQ4MzAwMCw5NzMwMDAsMjQ0OTAwMCwzNDc5MDAwLDM4ODkwMDAsMzk2MjAwMCw0MTMyMDAwLDM3NjkwMDAsMjc4NjAwMCwzOTUyMDAwLDI2MDgwMDAsNTExODAwMCw1OTEzMDAwLDM1MTQwMDAsNDYwODAwMCwzNTE3MDAwLDI4MjEwMDAsMjE3MTAwMCwzNzAxMDAwLDQwMjYwMDAsMTA3MzAwMCwzNzQxMDAwLDU0ODcwMDAsNDcxNzAwMCw0MzIwMDAsMTM3ODAwMCwzNjk5MDAwLDE4NDEwMDAsNTA0MDAwLDY5NTMwMDAsNTEyMzAwMCwzOTE5MDAwLDk0MDAwLDQ3MzEwMDAsMjYwOTAwMCwzNDQzMDAwLDUxNzAwMCwxNjgxMDAwLDE3ODcwMDAsNDkxMDAwMCwxNzUxMDAwLDIxMzkwMDAsMjQxMjAwMCw0OTQ4MDAwLDQwNzkwMDAsMzc1MDAwMCwyOTExMDAwLDI1NzgwMDAsNzMyMDAwMCwzNTMwMDAwLDE4ODkwMDAsMTgxODAwMCwzMDgxMDAwLDY0MTUwMDAsNTM3NzAwMCw0MzIzMDAwLDI0NjgwMDAsNTE4MTAwMCwyOTI2MDAwLDUxMjIwMDAsMzkxMDAwMCw1MjEzMDAwLDM5NzkwMDAsMzg0NjAwMCw1MTE0MDAwLDUxMjMwMDAsNDIwMDAwMCw0OTQxMDAwLDMzOTMwMDAsMTQ0MzAwMCwzODE0MDAwLDI0NDYwMDAsMzA3MTAwMCw0MDkyMDAwLDc2NDAwMCwxOTgzMDAwLDgwMzAwMCw1MDQzMDAwLDMwODAwMF0sIm9wcyI6MzI0MDg4MH1dLCJ1cGRhdGVkIjoiMjAyMy0wMy0yMVQwNzozOTo1NC43MThaIn0%3D). I'll replace it anyways but I would love to know why it isn't always faster.

Reply:
The regex is consistently faster on chrome on M1 🤔

Reply:
It seems `.includes` is consistently faster on my machine even if the regex is extracted. 🤔
I'm not sure about the reason though.